corgi/tasks/management/commands/updatecomponenttaxonomy.py

```diff
@@ -22,20 +29,49 @@ def handle(self, *args, **options):
                     f"updating {options['build_uuids']} component taxonomies",
                 )
             )
-            builds = SoftwareBuild.objects.filter(pk__in=options["build_uuids"])
+            for build in SoftwareBuild.objects.db_manager("read_only").filter(
+                pk__in=options["build_uuids"]
+            ):
+                self.stdout.write(
+                    self.style.SUCCESS(
+                        f"Saving product taxonomy for build: {build.build_type}:{build.build_id}"
+                    )
+                )
+                build.save_product_taxonomy()
+        elif options["product_ref"]:
+            for build_type, build_id in (
+                ProductComponentRelation.objects.db_manager("read_only")
+                .filter(product_ref=options["product_ref"])
+                .values_list("build_type", "build_id")
+                .distinct()
+                .iterator()
+            ):
+                self.stdout.write(
+                    self.style.SUCCESS(
+                        f"Saving product taxonomy for build: {build_type}:{build_id}"
+                    )
+                )
+                try:
+                    build = SoftwareBuild.objects.get(build_type=build_type, build_id=build_id)
+                except SoftwareBuild.DoesNotExist():
+                    self.stdout.write(
+                        self.style.WARNING(
+                            f"SoftwareBuild {build_type}:{build_id}, does not exists, so fetching."
+                        )
+                    )
+                    slow_fetch_modular_build.delay(build_type=build_type, build_id=build_id)
+                    continue
+                build.save_product_taxonomy()
         else:
             self.stdout.write(
                 self.style.SUCCESS(
                     "updating all builds component taxonomies",
                 )
             )
-            builds = SoftwareBuild.objects.get_queryset()
-        for sb in builds:
-            self.stdout.write(
-                self.style.SUCCESS(
-                    f"updating {sb.build_id}: {sb.name}",
+            for build in SoftwareBuild.objects.db_manager("read_only").all().iterator():
```

Comment:
If we want to let slow_fetch_brew_build handle the taxonomy-saving for us, and we don't need to call sb.save_product_taxonomy() directly, we can use .values_list("build_id", "build_type") to avoid fetching extra data from the DB:
```suggestion
            for build in SoftwareBuild.objects.db_manager("read_only").values_list("build_id", "build_type").iterator():
```
Or if .db_manager() doesn't support this method, you can do below instead:
```suggestion
            for build in SoftwareBuild.objects.values_list("build_id", "build_type").using("read_only").iterator():
```

But I think the old way was better, since that just saved the taxonomy directly instead of adding many tasks onto the slow queue. We could still add an .iterator() to the two queries there to avoid OoM issues when processing many builds or components.

Reply:
I removed this commit from the PR